dds/DCPS/DataReaderImpl.cpp

```diff
@@ -259,13 +259,15 @@ DataReaderImpl::add_association(const RepoId& yourId,
           writer_id,
           info));
 
-      // Schedule timer if necessary
-      //   - only need to check reader qos - we know the writer must be >= reader
-      if (this->qos_.durability.kind > DDS::VOLATILE_DURABILITY_QOS) {
-        info->waiting_for_end_historic_samples_ = true;
-      }
+    // Schedule timer if necessary
+    //   - only need to check reader qos - we know the writer must be >= reader
+    if (this->qos_.durability.kind > DDS::VOLATILE_DURABILITY_QOS) {
+      info->waiting_for_end_historic_samples_ = true;
+    }
+
+    remove_association_sweeper_->cancel_timer(writer_id);
```

Comment:
Related changes for the "sweeper" are already in master from #1723. I suggest merging master into this branch when work on this one resumes.